src/storage/mod.rs

```diff
@@ -78,10 +69,37 @@ pub enum StorageError {
     Unknown(String),
 }
 
+/// A trait that represents whether some struct can be read from a given store
+///
+/// Note that these msut be this way because we can't write non-overlapping blanked impls yet:
+/// https://github.com/rust-lang/rust/issues/20400
+/// https://stackoverflow.com/questions/40392524/conflicting-trait-implementations-even-though-associated-types-differ
+#[async_trait]
+pub trait StoreRead<State,Id> {
+    /// Read a given piece of state from the store
+    async fn read(&self, id: Id) -> Result<State, StorageError>;
+}
+
+/// A trait that represents whether some struct can be written to a given store
+#[async_trait]
+pub trait StoreWrite<State,Id> {
+    /// Write to a given store
+    async fn write(&mut self, obj: State, id: Option<Id>) -> Result<Id, StorageError>;
+}
+
+/// Trait for objects that can read and write some piece of state with a given identifier
+#[async_trait]
+pub trait StoreReadWrite<State,Id>: StoreRead<State,Id> + StoreWrite<State,Id> {}
+
 /// A trait that indicates the ability of a struct to store state
 /// Objects that implement Stores can save some given State, identifiable by StateId, and updatable with the given StateBuilder
+///
 #[async_trait]
-pub trait Stores {
+pub trait Stores
```

Comment:
Nit: I think most traits are singular

Reply:
I'm good with that!

- [ ] Change `Stores` to `Store`